src/lib.rs

```diff
@@ -104,6 +107,124 @@ impl Invoice {
     fn version_in_range(&self, requirement: &str) -> bool {
         version_compare(self.bindle.id.version(), requirement)
     }
+
+    fn cleartext(&self, by: String, role: SignatureRole) -> String {
+        let id = self.bindle.id.clone();
+        let mut buf = vec![
+            by.clone(),
+            id.name().to_owned(),
+            id.version_string(),
+            format!("{:?}", role).to_lowercase(),
```

Comment:
Nit: consider implementing `Display` instead on `SignatureRole` so that you can just use `to_string` rather than calling `Debug` (which I think is more expensive)

Reply:
I read up on this quite a bit before using `Debug`, and the advice was all to use Debug on enums. However, I just switched it to a Display implementation per your request. None of the arguments I could find really bothered to explain why one should prefer `Debug` except that it is more concise. So... meh.